### PR TITLE
feat: Scale up CE workflows

### PR DESCRIPTION
**Pull request by jamesclark-Zapata**
# The problem

CE workflows have been using the default resources.

# This PR's solution

* Use the requested resources in a workflow's task invocations to request a workflow's resources

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ check_untyped_defs = true
 
 [tool.coverage.run]
 # main.py should be checked separately in an integration test.
-omit = ["*/main.py", "*/__init__.py"]
+omit = ["*/main.py", "*/__init__.py", "src/orquestra/sdk/kubernetes/quantity.py"]
```
Comment (SebastianMorawiec) [inline pyproject.toml:41]: Nitpick: add comment that it is open sourced code that we have borrowed

File: src/orquestra/sdk/_base/_driver/_models.py [hunk 2]
```diff
@@ -198,13 +198,19 @@ def to_ir(self, workflow_def: WorkflowDef) -> WorkflowRun:
...
+class Resources(pydantic.BaseModel):
     """
     Implements:
-    https://github.com/zapatacomputing/workflow-driver/blob/2b3534/openapi/src/schemas/Runtime.yaml
+        https://github.com/zapatacomputing/workflow-driver/blob/6270a214fff40f53d7b25ec967f2e7875eb296e3/openapi/src/schemas/Resources.yaml
     """
 
-    SINGLE_NODE_RAY_RUNTIME = "singleNodeRayRuntime"
+    cpu: Optional[str] = pydantic.Field(
+        regex=r"^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$"
```
Comment (SebastianMorawiec) [inline src/orquestra/sdk/_base/_driver/_models.py:208]: Nonblocking: Oh noes, a long and complex regex. Could you somehow maybe document what it means? Same for GPU regex

**Review by SebastianMorawiec — COMMENTED** [review]
Code seems fine. I think it is worth to write 1 or 2 tests to actually test the logic of resource management

File: src/orquestra/sdk/_base/_driver/_models.py (hunk 2 above)
Comment (jamesclark-Zapata) [reply src/orquestra/sdk/_base/_driver/_models.py:208]: The regexes are from the spec: https://github.com/zapatacomputing/workflow-driver/blob/6270a214fff40f53d7b25ec967f2e7875eb296e3/openapi/src/schemas/Resources.yaml

This is in the docstring.

File: src/orquestra/sdk/_base/_driver/_ce_runtime.py [hunk 3]
```diff
@@ -86,10 +87,29 @@ def create_workflow_run(
         Returns:
             the workflow run ID
         """
+
+        max_gpu = None
+        max_memory = None
+        max_cpu = None
+        for inv in workflow_def.task_invocations.values():
```
[the remark this answers is not in the document]
Comment (jamesclark-Zapata) [reply src/orquestra/sdk/_base/_driver/_ce_runtime.py:94]: https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/76/commits/f67f184f496ed5cd586bda2f2acf71dd2853a117 wdyt?

**Review by SebastianMorawiec — APPROVED** [review]

**Pull request merged**